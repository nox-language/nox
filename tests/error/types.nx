struct Point {
    x: int,
    y: int,
}

fun main() = (
    if 12 then
        print "Hello\n";
    print 12;
    printi "hello";
    var point: Point = 12
)
