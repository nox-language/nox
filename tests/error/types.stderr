error: Unexpected type int32, expecting struct Point
   --> tests/error/types.nx:11:24
   |
11 |     var point: Point = 12
   |                        ^^

error: Unexpected type string, expecting int32
   --> tests/error/types.nx:10:12
   |
10 |     printi "hello";
   |            ^^^^^^^

error: Unexpected type int32, expecting string
   --> tests/error/types.nx:9:11
  |
9 |     print 12;
  |           ^^

error: Unexpected type int32, expecting bool
   --> tests/error/types.nx:7:8
  |
7 |     if 12 then
  |        ^^

